### ISSUE TITLE
[Tracking resistance] Nesting an encrypted SD-JWT combined presentation

### ISSUE THREAD

**Issue by peppelinux** [oauthstuff/draft-selective-disclosure-jwt#118]
Considering that SD-JWT makes selective disclosure and many privacy features possibile in the data format level, without any costraint in the transport protocol adopted during the exchange, I was wondering how to make a combined presentation unique and not trackable in environment where that can be the risk of eavesdropping (NFC, BT, QR codes, any other link of proximity ...)

Adopting a JWE and also a nesting strategy to a combined presentation can make the SD-JWT completely not trackable during the transport.

at first sight nesting a combined presentation in an encrypted JWT that MUST changes at every exchange makes the need to create an envelope JWT. It's a trade off, I know but an envelope JWT can cover also other aspects, like this

https://github.com/oauthstuff/draft-selective-disclosure-jwt/issues/114

**Comment by Sakurann** [comment]
Could you elaborate a little more what is being suggested? Is the idea to put multiple SD-JWT-Rs and SD-JWTs inside one JWE?
{header}.{
"SD-JWT-R-1": <...>
"SD-JWT-1": <...>
"SD-JWT-2": <...>
}.{signature}
kind of?

**Comment by peppelinux** [comment]
Yes, on each submission the JWE will be different (due to jti, iat) making it non trackable.
an adversary can't track how many time an SD-JWT have been submitted to an endpoint or on the wire, because the JWE will be alway unique

the proposal of a nested structure for a combined presentation have been proposed here

https://github.com/oauthstuff/draft-selective-disclosure-jwt/issues/114#issuecomment-1181556044

**Comment by peppelinux** [comment]
> Could you elaborate a little more what is being suggested? Is the idea to put multiple SD-JWT-Rs and SD-JWTs inside one JWE as claims? {header}.{ "SD-JWT-R-1": <...> "SD-JWT-1": <...> "SD-JWT-2": <...> }.{signature} kind of?

a simple array in a claim defined for this scope

````
"sd_vc": [ ... one or more SD-JWT ...],
````

**Comment by danielfett** [comment]
We discussed this at the IETF meeting in London. I think it is clear that we will not support multiple SD-JWTs in the format itself, as this opens up a whole new layer. For example, we would need to talk about relationships between SD-JWTs, what happens if only single credentials fail validation, etc.; all of this is not strictly related to the credential format and is much better handled on the transport protocol layer, e.g., OIDC4VC.

**Comment by TakahikoKawasaki** [comment]
Let me confirm my understanding (I'm a newbie to SD-JWT).

In the current snapshot of the SD-JWT specification, a binding JWT is placed at the end of "Combined Format for Presentation" like below.

    <SD-JWT>~<Disclosure 1>~<Disclosure 2>~...~<Disclosure M>~<optional Holder Binding JWT>

But the format is going to be discarded and something like below is going to be proposed or has already been proposed somewhere. Right?

**Payload of a binding JWT**

```json
{
  "{list of Combined Formats}": [
    "<SD-JWT-0>~<Disclosure-0-0>~...~<Disclosure-0-M>",
    "<SD-JWT-1>~<Disclosure-1-0>~...~<Disclosure-1-M>",
    "..."
  ]
}
```

**Comment by bc-pi** [comment]
Welcome to the SD-JWT party Taka :) 

The current Combined Format for Presentation isn't expected to change or be discarded. We do allow for an enveloping approach as in https://datatracker.ietf.org/doc/html/draft-ietf-oauth-selective-disclosure-jwt#section-7 and suggest the `_sd_jwt` claim be used for that. 

Encrypting isn't precluded but will likely be done/defined at an application protocol layer - i.e. JARM encryption of the vp_token(s) in OpenID4VP.

**Comment by Sakurann** [comment]
encrypting would be allowed by encrypting an entire JWT in the approach 2 in Taka's comment above
